### DISCUSSION TITLE
[Backend] Unison

### DISCUSSION THREAD

**Discussion by dantb** [realworld-apps/realworld#881] [WIP Implementations]
Language docs [here](https://www.unisonweb.org/docs). Excited to try it out in anger, in particular the effect system (abilities AKA algebraic effect handlers in the formal literature).

Unison is in alpha with a small library ecosystem at the moment. So it might require adding to that ecosystem to facilitate this app.

Will update this with the repo at a later date (since definitions in code are content-addressed, the codebase is not human readable but rather a sqlite database of hashes / metadata. You need to use `ucm` to view it).

If anyone has done / wants to do something similar in Unison I'd be happy to collaborate 🙂

**Comment by rashadg1030** [comment]
Are using Frog for this? Last time I checked that was the only web framework for Unison, but it was still work in progress last time I checked. Are you just gonna use the low-level HTTP library? I'm not even sure that was fully implemented either, but I haven't checked recently.

**Comment by dantb** [comment]
Yeah I've been using the lower level [unison.http](https://share.unison-lang.org/latest/namespaces/unison/http) directly. Perhaps it will get unwieldy and call for something like Frog. Have you been down a similar path @rashadg1030 ?

**Reply by rashadg1030** [reply]
> Yeah I've been using the lower level [unison.http](https://share.unison-lang.org/latest/namespaces/unison/http) directly. Perhaps it will get unwieldy and call for something like Frog. Have you been down a similar path @rashadg1030 ?

Oh okay. No, I haven't. I've been following Unison for some time and I remember asking in the Slack channel if Unison was useable for web development. I was told about Frog, which is supposed to be like the Ruby Rack framework, but it wasn't ready at the time.

Checking now, it seems like https://share.unison-lang.org/latest/namespaces/frog/frog is still WIP so I think it's the right move to use the http library for now. Anyways, I'd like to help out. Is there a link to the repo you're working on?

**Reply by dantb** [reply]
Apologies for the delay @rashadg1030 , repo is [here](https://github.com/dantb/unison-realworld-example-app) and the project board is [here](https://github.com/users/dantb/projects/2). It's pretty rough around the edges. Should be on Unison share soon

**Comment by err0r500** [comment]
Nice ! 
I had to have a look at your profile in order to find the repo, so here it is in case someone also wants to have a look / give a hand : https://github.com/dantb/unison-realworld-example-app

**Comment by dantb** [comment]
Yeah sorry been meaning to update this! Thanks @err0r500

**Comment by dantb** [comment]
Also got a project board - https://github.com/users/dantb/projects/2

**Comment by rlmark** [comment]
Hey @dantb, this is Rebecca, from the Unison team. Are you still working on this? At the time, Unison was in its infancy and lacked a full HTTP framework and Postgres client, likely hampering your efforts. I'm happy to report we have both now! I'm doing some work on a real-world app implementation in Unison this week; if you'd like to join in, I'd be happy to combine efforts!

**Reply by dantb** [reply]
Hi Rebecca, no I haven't touched it in years (should've updated this since it wasn't actually WIP!). I don't think I got particularly far - I remember going down a rabbit hole implementing RSA for JWT to support user authentication. I was working to a thesis deadline and ended up going in a different direction (effect systems in Haskell). Anyway, it looks like there are libraries for all that good stuff now.

Whatever you are working on should definitely be the "official" Unison RealWorld app :) I'll close this discussion to avoid any confusion.
